### DISCUSSION TITLE
Import existing app module into a custom interface

### DISCUSSION THREAD

**Discussion by dmtkpv** [directus/directus#3576] [Feature Requests]
Directus has `@tinymce/tinymce-vue` dependency

I would like to use it in a custom interface. But import like this
```javascript
import Editor from '@tinymce/tinymce-vue';
```
throws an error
```
TypeError: Failed to resolve module specifier "@tinymce/tinymce-vue". Relative references must start with either "/", "./", or "../".
```
It would be great to have the ability to import existing modules into a custom interface

**Comment by rijkvanzanten** [comment]
Extensions aren't bundled in during build-time, so there's no way to import the dependencies that the app are using directly like that. The only way to re-use existing dependencies of the app is when the app itself registers them as globals (either on window or through Vue's provide/inject). As of right now, the Vue components that the app uses are all registered globally (so things like `v-button` work in extensions), but the npm dependencies aren't.

You'll have to re-install `@tinymce/tinymce-vue` in your custom interface 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
